### PR TITLE
chore(storage-browser): add refresh button to locations view

### PR DESCRIPTION
**Pull request by timngyn**
#### Description of changes
- For now, added `onClick` and `disabled` props for `RefreshControl` component
- Created custom `LocationViewsRefresh` component to handle refreshing locations view

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hbuchel — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/Views/LocationsView/LocationsView.tsx [hunk 1]
```diff
@@ -23,10 +24,26 @@ export interface LocationsView<
...
+
+  return (
+    <Refresh
+      disabled={isLoading}
+      onClick={() =>
+        handleListLocations({
+          options: { refresh: true, pageSize: 1000 },
+        })
+      }
+    ></Refresh>
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/Views/LocationsView/LocationsView.tsx:38]:
```suggestion
    <Refresh
      disabled={isLoading}
      onClick={() =>
        handleListLocations({
          options: { refresh: true, pageSize: 1000 },
        })
      }
    />
```

**Review by hbuchel — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/Views/LocationsView/LocationsView.tsx [hunk 2]
```diff
@@ -23,10 +24,26 @@ export interface LocationsView<
   T extends StorageBrowserElements = StorageBrowserElements,
 > extends ViewComponent<LocationsViewControls<T>> {}
 
+const LocationsViewRefresh = () => {
+  const [{ isLoading }, handleListLocations] = useLocationsData();
+
+  return (
+    <Refresh
+      disabled={isLoading}
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/Views/LocationsView/LocationsView.tsx:32]: Let's also disable here when there is no data, e.g. `data.result.length <= 0`

**Review by hbuchel — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/Views/Controls/Refresh.tsx [hunk 3]
```diff
@@ -34,8 +34,8 @@ const RefreshButton = withBaseElementProps(Button, {
   variant: 'refresh',
 });
 
-export const RefreshControl: RefreshControl = () => (
-  <RefreshButton>
+export const RefreshControl: RefreshControl = ({ onClick, disabled }) => (
+  <RefreshButton onClick={onClick} {...(disabled ? { disabled: true } : {})}>
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/Views/Controls/Refresh.tsx:38]: Maybe missing some context here but think we can avoid the spread logic?
```suggestion
export const RefreshControl: RefreshControl = ({ onClick, disabled = false }) => (
  <RefreshButton onClick={onClick} disabled={disabled}>
```

Comment (timngyn) [reply packages/react-storage/src/components/StorageBrowser/Views/Controls/Refresh.tsx:38]: oh thanks Caleb, for some reason last time I tried it locally, I was getting an eslint error about the value needing to be omitted if it was boolean. I just tried your changes locally and I don't seem to get it anymore so maybe I just set up something incorrectly the first time 🤦

**Review by calebpollman — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

**Pull request merged**